### PR TITLE
Validate entries in ALLOWLIST are not owned

### LINKED ISSUE #12629: [check-codeowners.sh] Validate entries in ALLOWLIST are not owned
The `.github/scripts/check-codeowners.sh` script is useful to validate that components have an owner by checking that all entries on CODEOWNERS are valid and are not in the ALLOWLIST file.

It would be useful to also validate that the ALLOWLIST file does not have any entries that actually have an owner, so that it accurately reflects the orphaned components.

### PR DESCRIPTION
**Pull request by Chinwendu20**
Signed-off-by: Maureen <amaka013@gmail.com>

**Description:** I created a function in .github/workflows/scripts/check-codeowners.sh file that validates if the modules in the OWNERLIST are owned, it prints out the modules in ownerlist that are contained in the codeowners file.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/12629

### CODE REVIEW

**Review by mx-psi — CHANGES_REQUESTED** [review]
You will also need to call the script on the Github Actions workflow definition. Here is the current place where we call the other functions:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/75234802c667663092b85eb15b32a0aec06538dc/.github/workflows/build-and-test.yml#L58-L61

**Comment by Chinwendu20** [comment]
> You will also need to call the script on the Github Actions workflow definition. Here is the current place where we call the other functions:
> 
> https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/75234802c667663092b85eb15b32a0aec06538dc/.github/workflows/build-and-test.yml#L58-L61

Okay, thanks

**Comment by Chinwendu20** [comment]
It is failing because there are non-orphaned components in the ALLOWLIST file:
https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3281736897/jobs/5404313005

**Comment by mx-psi** [comment]
> It is failing because there are non-orphaned components in the ALLOWLIST file: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3281736897/jobs/5404313005

This is good, it means the check is working :) 

Could you remove the offending ALLOWLIST lines on this PR as well to make the check pass?

**Comment by Chinwendu20** [comment]
> > It is failing because there are non-orphaned components in the ALLOWLIST file: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3281736897/jobs/5404313005
> 
> This is good, it means the check is working :)
> 
> Could you remove the offending ALLOWLIST lines on this PR as well to make the check pass?

Okay I have removed the offending lines in the ALLOWLIST file

File: .github/workflows/scripts/check-codeowners.sh [hunk 1]
```diff
@@ -86,8 +86,28 @@ check_component_existence() {
...
+  fi
+}
+
 if [[ "$1" == "check_code_owner_existence" ]];  then
   check_code_owner_existence
 elif [[ "$1" == "check_component_existence" ]]; then
   check_component_existence
+elif [[ "$1" == "check_entries_in_allowlist" ]]; then
+  check_entries_in_allowlist
 fi
```
Comment (TylerHelmuth) [inline .github/workflows/scripts/check-codeowners.sh:113]: Can you add a newline to the end of this file?

**Review by TylerHelmuth — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

File: .github/workflows/scripts/check-codeowners.sh (hunk 1 above)
Comment (Chinwendu20) [reply .github/workflows/scripts/check-codeowners.sh:113]: Oh ok. Thanks

Comment (Chinwendu20) [reply .github/workflows/scripts/check-codeowners.sh]: I have done that @TylerHelmuth

**Pull request merged**